### PR TITLE
Bug 2037209: [Alibaba] refactor for getResourceGroup ID from name

### PR DESCRIPTION
**Pull request by menglingwei**

### CODE REVIEW

File: pkg/actuators/machine/instances.go [hunk 1]
```diff
@@ -824,16 +804,44 @@ func correctExistingTags(machine *machinev1beta1.Machine, regionID string, insta
...
+			Name:      machine.Name,
+			Namespace: machine.Namespace,
+			Reason:    err.Error(),
+		})
+		klog.Errorf("error list resourcegroups: %v", err)
+		return "", fmt.Errorf("error list resourcegroups: %v", err)
+	}
+	if len(response.ResourceGroups.ResourceGroup) < 1 {
+		klog.Errorf("no resourcegroups for given name not found")
+		return "", fmt.Errorf("no resourcegroups for given name not found")
```
Comment (elmiko) [inline pkg/actuators/machine/instances.go:843]: minor nit
```
"no resourcegroups for given name found"
```

**Review by elmiko — COMMENTED** [review]
this generally looks good to me, just a minor cleanup in the error message

@JoelSpeed ptal

File: pkg/actuators/machine/instances.go (hunk 1 above)
Comment (menglingwei) [reply pkg/actuators/machine/instances.go]: fixed it

**Review by elmiko — COMMENTED** [review]
/lgtm

**Pull request merged**